.github/workflows/android.yml

```diff
@@ -9,64 +9,23 @@ on:
     branches: [master, staging, next]
 
   workflow_dispatch:
+    inputs:
+      release:
+        description: 'release type'
+        required: true
+        default: 'qa'
+      targetbranch:
+        description: 'target deployment branch'
+        required: false
+        default: 'staging'
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
-  test:
-    name: Detox Tests
-    runs-on: ubuntu-latest
-    #disabled until we get it working again
-    if: ${{ false && !contains(github.event.head_commit.message, 'skip ci') }}
-    steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v1
-        with:
-          node-version: 14
-      - name: Use specific Java version for sdkmanager to work
-        uses: joschi/setup-jdk@v1
-        with:
-          java-version: 'openjdk11'
-          architecture: 'x64'
-      - name: Install npm dependencies
-        uses: bahmutov/npm-install@v1
-        with:
-          install-command: yarn --frozen-lockfile
-      - name: Cache Android
-        id: cache-android
-        uses: actions/cache@v1
-        with:
-          path: .android
-          key: ${{ runner.os }}-android-29
-      - name: Download Android Emulator Image
-        if: ${{ steps.cache-android.outputs.cache-hit != 'true' }}
-        run: |
-          echo "y" | $ANDROID_HOME/tools/bin/sdkmanager --install "system-images;android-29;google_apis;x86"
-          echo "no" | $ANDROID_HOME/tools/bin/avdmanager create avd --force --name Nexus_5X_API_29_x86 --device "Nexus 5X" -k 'system-images;android-29;google_apis;x86'
-      # - name: Android Emulator
-      #   timeout-minutes: 10
-      #   continue-on-error: true
-      #   run: |
-      #     echo "Starting emulator"
-      #     $ANDROID_HOME/emulator/emulator -list-avds
-      #     nohup $ANDROID_HOME/emulator/emulator -avd emu -no-audio -no-snapshot -no-window &
-      #     $ANDROID_HOME/platform-tools/adb wait-for-device shell 'while [[ -z $(getprop sys.boot_completed | tr -d '\r') ]]; do sleep 1; done; input keyevent 82'
-      #     $ANDROID_HOME/platform-tools/adb devices
-      #     echo "Emulator started"
-      - name: Prepare env
-        env: # Or as an environment variable
-          SENTRYRC: ${{ secrets.sentryrc_file }}
-        run: |
-          echo "$SENTRYRC" > android/sentry.properties
-      - name: run detox build & test
-        run: |
-          $ANDROID_HOME/emulator/emulator -list-avds
-          yarn lingui-compile
-          yarn test:android
   # This workflow contains a single job called "build"
   build:
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
-    if: "!contains(github.event.head_commit.message,'release:') && !contains(github.event.head_commit.message,'skip ci')"
+    if: ${{ endsWith(github.ref, '/master') || !contains(github.event.head_commit.message,'release:') && !contains(github.event.head_commit.message,'skip ci') }}
```

Comment:
condition to run **full release** = `<manual run>` || `<release run>` || `<PR merged>`

`<PR merged>` = <branch **is** _master_> && <commit message **ends with** _`/\(\d+\)/`_>
`<release run>` = (<branch **is** _staging_> && <commit message **contains** _'release qa'_>) || (<branch **is** _next_> && <commit message **contains** _'release prod'_>)

Reply:
you could keep github.event.pull_request.merged == true for 'PR merged' condition